scripts/junit-to-markdown.py

```
@@ -0,0 +1,142 @@
+The test-report job collects all rows and builds a single markdown table.
+"""
+
+import json
+import re
+import sys
+import xml.etree.ElementTree as ET
+from pathlib import Path
+
+
+def parse_junit(xml_path: str) -> dict:
+    tree = ET.parse(xml_path)
```

Ruff (0.15.21)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)